tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1865 @@
+
+        :param cmd: The command to execute.
+        :param timeout: The command timeout.
+        :param silent: Whether to run the command silent without logging.
+        :returns: The command output as a list of lines.
+        """
+        if not silent:
+            self.__logger and self.__logger.info('> %s', cmd)
+
+        output = self.__otcmd.execute_command(cmd, timeout)
+
+        if not silent:
+            for line in output:
+                self.__logger and self.__logger.info('%s', line)
+
+        if cmd in ('reset', 'factoryreset'):
+            self.__wait_reset()
+            return output
+
+        if output[-1] == 'Done':
+            output = output[:-1]
+            return output
+        else:
+            raise CommandError(cmd, output)
+
+    def set_logger(self, logger: logging.Logger):
+        """Set the logger for the OTCI instance, or None to disable logging."""
+        self.__logger = logger
+
+    #
+    # Constant properties
+    #
+    @constant_property
+    def version(self):
+        """Returns the firmware version. (e.g. "OPENTHREAD/20191113-01411-gb2d66e424-dirty; SIMULATION; Nov 14 2020 14:24:38")"""
+        return self.__parse_str(self.execute_command('version'))
+
+    @constant_property
+    def thread_version(self):
+        """Get the Thread Version number."""
+        return self.__parse_int(self.execute_command('thread version'))
+
+    @constant_property
+    def api_version(self):
+        """Get API version number."""
+        try:
+            return self.__parse_int(self.execute_command('version api'))
+        except ValueError:
+            # If the device does not have `version api` command, it will print the firmware version, which would lead to ValueError.
+            return 0
+
+    #
+    # Basic device operations
+    #
+    def ifconfig_up(self):
+        """Bring up the IPv6 interface."""
+        self.execute_command('ifconfig up')
+
+    def ifconfig_down(self):
+        """Bring down the IPv6 interface."""
+        self.execute_command('ifconfig down')
+
+    def get_ifconfig_state(self) -> bool:
+        """Get the status of the IPv6 interface."""
+        return self.__parse_values(self.execute_command('ifconfig'), up=True, down=False)
+
+    def thread_start(self):
+        """Enable Thread protocol operation and attach to a Thread network."""
+        self.execute_command('thread start')
+
+    def thread_stop(self):
+        """Disable Thread protocol operation and detach from a Thread network."""
+        self.execute_command('thread stop')
+
+    def reset(self):
+        """Signal a platform reset."""
+        self.execute_command('reset')
+
+    def factory_reset(self):
+        """Delete all stored settings, and signal a platform reset."""
+        self.execute_command('factoryreset')
+
+    #
+    # Network Operations
+    #
+
+    def ping(self, ip: str, size: int = None, count: int = None, interval: int = None, hoplimit: int = None):
```

Comment:
Is there plan for similar functionality of  ping() method in existing Node.py, which has `num_responses` and `timeout` parameters?
https://github.com/openthread/openthread/blob/8630547d106daeaff64c6c960ff316036222e4db/tests/scripts/thread-cert/node.py#L1327

Reply:
We already have `count` and `interval` which is same as `num_responses` and `timeout`. \
OTCI is following the argument names used in https://github.com/openthread/openthread/blob/master/src/cli/README.md#ping-ipaddr-sizecount-intervalhoplimit